blog/urls.py

```diff
@@ -0,0 +1,8 @@
+from . import views
+from django.urls import path
+
+urlpatterns = [
+    path('', views.PostList.as_view(), name='home'),
+    path('<slug:slug>/', views.PostDetail.as_view(), name='post_detail'),
+    path('',views.index, name="homepage"),
```

Comment:
Why do you have two url paths that map to two different views, i.e. why does `''` map to `views.PostList.as_view()` and `views.index()` ? Seems like it would be prone to some sort of issue.